iam_groups_authn/database.py

```diff
@@ -41,13 +49,15 @@ def mysql_username(iam_email):
 class RoleService:
     """Class for managing a DB user's role grants."""
 
-    def __init__(self, db):
+    def __init__(self, db, database_type):
```

Comment:
Essentially, every single function on this object is using database_type to define behavior. 

What's cleaner instead is if you have a "MysqlRoleService" and a "PostgresRoleService" that each have their define operations. They can share a "RoleService" interface to enforce typing.